python/hail/utils/misc.py

```diff
@@ -17,19 +20,157 @@ def wrap_to_list(s):
     else:
         return [s]
 
+
 def get_env_or_default(maybe, envvar, default):
     import os
 
     return maybe or os.environ.get(envvar) or default
 
+
 def get_URI(path):
     return Env.jutils().getURI(path)
 
+
 @handle_py4j
-def new_temp_file(n_char = 10, prefix=None, suffix=None):
+def new_temp_file(n_char=10, prefix=None, suffix=None):
     return Env.hc()._jhc.getTemporaryFile(n_char, joption(prefix), joption(suffix))
 
+
 storage_level = enumeration('NONE', 'DISK_ONLY', 'DISK_ONLY_2', 'MEMORY_ONLY',
                             'MEMORY_ONLY_2', 'MEMORY_ONLY_SER', 'MEMORY_ONLY_SER_2',
                             'MEMORY_AND_DISK', 'MEMORY_AND_DISK_2', 'MEMORY_AND_DISK_SER',
-                            'MEMORY_AND_DISK_SER_2', 'OFF_HEAP')
\ No newline at end of file
+                            'MEMORY_AND_DISK_SER_2', 'OFF_HEAP')
+
+
+def plural(orig, n, alternate=None):
+    if n == 1:
+        return orig
+    elif alternate:
+        return alternate
+    else:
+        return orig + 's'
+
+
+def get_nice_attr_error(obj, item):
+    from hail.api2 import MatrixTable, GroupedMatrixTable, Table, GroupedTable
+    if isinstance(obj, MatrixTable):
+        class_name = 'MatrixTable'
+        cls = MatrixTable
+        index_obj = obj
+    elif isinstance(obj, GroupedMatrixTable):
+        class_name = 'GroupedMatrixTable'
+        cls = GroupedMatrixTable
+        index_obj = obj._parent
+    elif isinstance(obj, Table):
+        class_name = 'Table'
+        cls = Table
+        index_obj = obj
+    elif isinstance(obj, GroupedTable):
+        class_name = 'GroupedTable'
+        cls = GroupedTable
+        index_obj = obj._parent
+    else:
+        raise NotImplementedError(obj)
+
+    if item.startswith('_'):
```

Comment:
We don't forbid users from using fields that start with "_". This will break if someone says `table.annotate(**{'_foo': table.X + 5})`

Reply:
it's only called on a keyerror, so underscore-prefixed fields will work as before (just won't get nice similarity guessing if you do `ds._foo`.